tests/src/test/scala-3/pureconfig/ProductReaderDerivationSuite.scala

```diff
@@ -0,0 +1,99 @@
+package pureconfig
+package generic
+
+import scala.collection.JavaConverters.given
+import scala.language.higherKinds
+
+import com.typesafe.config.{ConfigFactory, ConfigRenderOptions, ConfigValueFactory}
+import org.scalacheck.Arbitrary
+import pureconfig._
+import pureconfig.ConfigConvert.catchReadError
+import pureconfig.error.{KeyNotFound, WrongType}
+import pureconfig.generic.derivation.default.derived
+
+class ProductReaderDerivationSuite extends BaseSuite {
```

Comment:
I think there are two tests missing here:

- A simple success case, where we have a case class with 2-3 primitive or collection parameters and we check that it is read successfully;
- A test verifying that we can read config lists from tuples.

Reply:
The last test should verify that `{ values: [ true, 5] }` can be read as `case class Foo(values: (Boolean, Int))`, if I understand you correctly?

Reply:
Yes, that's correct!